Framework/PythonInterface/plugins/algorithms/roundinghelper.py

```diff
@@ -0,0 +1,37 @@
+from __future__ import (absolute_import, division, print_function)
+from mantid.kernel import Direction, StringListValidator
+import numpy
+
+'''
+This file contains functions which deal with rounding in algorithms
+such as BinWidthAtX and MedianBinWidth.
+
+'''
+
+# Name of the rounding mode property
+PROP_NAME_ROUNDING_MODE = 'Rounding'
+
+# Available rounding modes.
+ROUNDING_NONE = 'None'
+ROUNDING_TEN_TO_INT = '10^n'
+
+
+def declare_rounding_property(o):
+    '''
+    Declares the properties needed for rounding.
+    '''
+    rounding = StringListValidator()
+    rounding.addAllowedValue(ROUNDING_NONE)
+    rounding.addAllowedValue(ROUNDING_TEN_TO_INT)
+    o.declareProperty(name=PROP_NAME_ROUNDING_MODE,
+                      defaultValue=ROUNDING_NONE, validator=rounding,
+                      direction=Direction.Input, doc='Bin width rounding')
+
+
+def round(x, mode):
+    '''
+    Rounds x depending on the rounding mode selected.
+    '''
+    if mode == ROUNDING_TEN_TO_INT:
+        return 10.0**numpy.floor(numpy.log10(x))
```

Comment:
Indeed. This brought me to another point, that these algorithms should be agnostic to the ordering of the bin boundaries, i.e. give meaningful results regardless of whether the x axis is ascending or descending. Unfortunately, turns out that `binIndexOf()` in `MatrixWorkspace` doesn't work with descending binning neither.

I opened an issue #18020 for this. I think I need to fix that issue before I can finish this one.